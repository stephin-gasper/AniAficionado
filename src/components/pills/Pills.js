import {ScrollView} from 'react-native';
import React, {memo, useState} from 'react';
import PropTypes from 'prop-types';
import {useTheme} from 'styled-components/native';

import {Pill, PillText, styles} from './Pills.style';

const Pills = memo(({list, onPillPress}) => {
  const theme = useTheme();
  const [activePillIndex, setActivePillIndex] = useState(0);

  return (
    <ScrollView
      showsVerticalScrollIndicator={false}
      horizontal
      contentContainerStyle={styles.contentContainer}>
      <For each="text" index="index" of={list}>
        <Pill
          key={`pill-${text}`}
          initialColor={
            index === 0 ? theme.tertiaryColor : theme.secondaryColor
          }
          activeColor={
            index === activePillIndex
              ? theme.tertiaryColor
              : theme.secondaryColor
          }
          testID={`pill-${index}`}>
          <PillText
            onPress={() => {
              setActivePillIndex(index);
              onPillPress(text);
            }}
            testID={`pill-text-${index}`}>
            {text}
          </PillText>
        </Pill>
      </For>
    </ScrollView>
  );
});

Pills.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string).isRequired,
  onPillPress: PropTypes.func.isRequired,
};

export default Pills;
