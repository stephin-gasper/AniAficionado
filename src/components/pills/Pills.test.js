import React from 'react';
import {View as MockView} from 'react-native';
import {render, act, fireEvent} from 'test/test-utils';

import Pills from './Pills';

jest.mock('components/animated-background-color-view', () => props => (
  <MockView {...props} />
));

describe('<Pills />', () => {
  const mockOnPillPress = jest.fn();
  const defaultProps = {
    list: ['pill1', 'pill2'],
    onPillPress: mockOnPillPress,
  };

  it('should render correctly', () => {
    expect(render(<Pills {...defaultProps} />)).toMatchSnapshot();
  });

  it('should change active pill on click of pill', async () => {
    const {getByTestId} = render(<Pills {...defaultProps} />);

    expect(getByTestId('pill-0').props.activeColor).toBe('#352462');
    expect(getByTestId('pill-1').props.activeColor).toBe('#9381b9');

    await act(async () => {
      fireEvent(getByTestId('pill-text-1'), 'onPress');
    });

    expect(getByTestId('pill-0').props.activeColor).toBe('#9381b9');
    expect(getByTestId('pill-1').props.activeColor).toBe('#352462');

    await act(async () => {
      fireEvent(getByTestId('pill-text-0'), 'onPress');
    });

    expect(getByTestId('pill-0').props.activeColor).toBe('#352462');
    expect(getByTestId('pill-1').props.activeColor).toBe('#9381b9');

    expect(mockOnPillPress).toHaveBeenCalledTimes(2);
  });
});
