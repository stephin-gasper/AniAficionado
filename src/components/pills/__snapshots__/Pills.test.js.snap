// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pills /> should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "marginBottom": 20,
      "marginHorizontal": -3,
    }
  }
  horizontal={true}
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      activeColor="#352462"
      initialColor="#352462"
      style={
        [
          {
            "borderRadius": 20,
            "marginBottom": 0,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 0,
          },
        ]
      }
      testID="pill-0"
    >
      <Text
        onPress={[Function]}
        style={
          [
            {
              "color": "#fffade",
              "paddingBottom": 6,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 6,
            },
          ]
        }
        testID="pill-text-0"
      >
        pill1
      </Text>
    </View>
    <View
      activeColor="#9381b9"
      initialColor="#9381b9"
      style={
        [
          {
            "borderRadius": 20,
            "marginBottom": 0,
            "marginLeft": 3,
            "marginRight": 3,
            "marginTop": 0,
          },
        ]
      }
      testID="pill-1"
    >
      <Text
        onPress={[Function]}
        style={
          [
            {
              "color": "#fffade",
              "paddingBottom": 6,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 6,
            },
          ]
        }
        testID="pill-text-1"
      >
        pill2
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
